libnmstate/nmstate.py

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (c) 2018-2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+
+from contextlib import contextmanager
+
+from libnmstate import validator
+from libnmstate.nm import NetworkManagerPlugin
+from libnmstate.schema import DNS
+from libnmstate.schema import Interface
+from libnmstate.schema import Route
+from libnmstate.schema import RouteRule
+
+
+@contextmanager
+def plugin_context():
+    nm_plugin = NetworkManagerPlugin()
+    try:
+        yield nm_plugin
+    finally:
+        nm_plugin.unload()
+
+
+def show_with_plugin(plugin, include_status_data=None):
+    plugin.refresh_content()
+    report = {}
+    if include_status_data:
+        report["capabilities"] = plugin.capabilities
+
+    report[Interface.KEY] = plugin.get_interfaces()
+    report[Route.KEY] = plugin.get_routes()
+    report[RouteRule.KEY] = plugin.get_route_rules()
+    report[DNS.KEY] = plugin.get_dns_client_config()
+
+    validator.validate(report)
```

Comment:
Does this make sense? I think we should trust the output. If this fails it means the user has been able to apply a wrong state. What do you think?

Reply:
Does this make sense? This will fail if the report does not fit our schema from `libnmstate/schemas/operational-state.yaml`. What would happen if an user has a interface that we are not supporting yet? Will he not be allowed to do `nmstatectl show`?

Reply:
This is the original behavior and unrelated to this PR.

Reply:
Okay, seems fair then.